src/vector/app.js

```diff
@@ -54,11 +54,25 @@ import CallHandler from 'matrix-react-sdk/src/CallHandler';
 let lastLocationHashSet = null;
 
 function checkBrowserFeatures(featureList) {
+    let featureComplete = true;
+
+    // custom checks atop Modernizr because it doesn't have ES2018/ES2019 checks in it for some features we depend on:
+    // ran prior to Modernizr so the missing features are logged even if Modernizr is broken
+    // ES2018: http://www.ecma-international.org/ecma-262/9.0/#sec-promise.prototype.finally
+    if (typeof Promise.prototype.finally !== "function") {
```

Comment:
Hmm, I think this needs to check more carefully, since `Promises` as a whole may also be missing. Either use the Modernizr promises check and move this after, or change this to test for `Promises` first, then `finally`.

Reply:
Fair point!